sdk/nodejs/runtime/rpc.ts

```diff
@@ -601,3 +603,29 @@ export function registerResourcePackage(pkgName: string, version: string, pkg: R
     }
     resourcePackages.set(key, pkg);
 }
+
+/**
+ * A ResourceModule is a type that understands how to construct resources given a name, type, args, and URN.
+ */
+export type ResourceModule = {
```

Comment:
Nit: Any reason this is a type alias rather than an interface? Just wondering if it could be an interface (same for `ResourcePackage`). https://www.typescriptlang.org/docs/handbook/advanced-types.html#interfaces-vs-type-aliases

Reply:
Yes, we should probably make these interfaces. Not sure why I used a type alias, tbh.